be/tool/beck.c

```diff
@@ -511,6 +521,31 @@ int main(int argc, char **argv)
 		      err(EX_NOINPUT, "Cannot seek snapshot to the beginning.");
 		printf("Snapshot size: %"PRId64".\n", s.s_size);
 	}
+
+	generation_id_get(s.s_file, &gen_id);
+	if (print_gen_id) {
+		generation_id_print(gen_id);
+		printf("\n");
+		fclose(s.s_file);
+		return EX_OK;
+	}
+	if (s.s_gen != 0) {
+		s.s_gen_found = true;
+		printf("\nReceived source segment header generation id\n");
+		generation_id_print(s.s_gen);
+	}
+
+	s.s_gen = gen_id ?: s.s_gen;
```

Comment:
[* question *] Do I understand correctly that it's not possible to override `gen_id` if it was found in the segment? If so, then it seems like a bug.

Reply:
@max-seagate  - Yes the idea was if 'gen_id; is found then use it else fallback to whatever is present in s.s_gen. In short' gen_id' overrides s.s_gen.